### PR TITLE
Add curl port support and post hooking

### PR DESCRIPTION
**Pull request by tudor-timcu**
- Added generic support for hooking handlers for both pre-execution and post-execution of a specific PHP function
- Hooked curl_exec in post-execution and extracted the destination port
- Pass domain + port to the agent
- Agent forwards the port to cloud on heartbeat events

### CODE REVIEW

File: tests/cli/test_curl.phpt [hunk 1]
```diff
@@ -35,6 +35,9 @@ curl_close($ch3);
...
 --EXPECT--
-[AIKIDO][INFO] Got domain: example.com
-[AIKIDO][INFO] Got domain: httpbin.org
-[AIKIDO][INFO] Got domain: facebook.com
+[AIKIDO][INFO] [BEFORE] Got domain: example.com
+[AIKIDO][INFO] [AFTER] Got domain: example.com port: 443
+[AIKIDO][INFO] [BEFORE] Got domain: httpbin.org
+[AIKIDO][INFO] [AFTER] Got domain: httpbin.org port: 443
+[AIKIDO][INFO] [BEFORE] Got domain: facebook.com
+[AIKIDO][INFO] [AFTER] Got domain: facebook.com port: 443
```
Comment (hansott) [inline tests/cli/test_curl.phpt:59]: might be good to add the following cases:
* `curl_init("https://facebook.com:443")`
* `curl_init("http://www.aikido.dev:80")`
* `curl_init("http://domain-with-random-port:4113")`

File: lib/php-extension/HandleUrls.cpp [hunk 2]
```diff
@@ -27,23 +27,68 @@ AIKIDO_HANDLER_FUNCTION(handle_curl_exec) {
...
+	zval* fname = NULL;
+
+	fname = (zval*)emalloc(sizeof(zval));
+	if (fname == NULL) {
+		return;
+	}
+		
+	ZVAL_STRING(fname, "curl_getinfo");
+
+	if (call_user_function(EG(function_table), NULL, fname, &retval, 2, params) != SUCCESS) {
```
Comment (willem-delbare) [inline lib/php-extension/HandleUrls.cpp:81]: best add some comments as to what we are doing here.

File: lib/php-extension/HandleUrls.cpp [hunk 3]
```diff
@@ -27,23 +27,68 @@ AIKIDO_HANDLER_FUNCTION(handle_curl_exec) {
...
+		return;
+	}
+	if (Z_TYPE(retval) != IS_LONG) {
```
Comment (willem-delbare) [inline lib/php-extension/HandleUrls.cpp:84]: what does this mean? why do we not need to (e)free the malloc'd var anymore? the early returns are not a memleak?

File: lib/php-extension/HandleUrls.cpp [hunk 4]
```diff
@@ -27,23 +27,68 @@ AIKIDO_HANDLER_FUNCTION(handle_curl_exec) {
...
+
+	zval retval;
+	zval params[2];
+	ZVAL_COPY(&params[0], curlHandle);
+	ZVAL_LONG(&params[1], CURLINFO_PRIMARY_PORT);
+	zval* fname = NULL;
+
+	fname = (zval*)emalloc(sizeof(zval));
+	if (fname == NULL) {
+		return;
```
Comment (willem-delbare) [inline lib/php-extension/HandleUrls.cpp:76]: not a memleak?

File: lib/request-processor/handle_urls.go [hunk 5]
```diff
@@ -6,13 +6,28 @@ import (
...
+func OnBeforeFunctionExecutedCurl(parameters map[string]interface{}) string {
 	url := utils.GetFromMap[string](parameters, "url")
 	if url == nil {
 		return "{}"
 	}
 	domain := utils.GetDomain(*url)
-	log.Info("Got domain: ", domain)
-	go grpc.OnDomain(domain)
+	log.Info("[BEFORE] Got domain: ", domain)
+	//TODO: check if domain is blacklisted
```
Comment (willem-delbare) [inline lib/request-processor/handle_urls.go:16]: best craete jira tickets, don't use TODO

**Pull request merged**